packages/dd-trace/src/debugger/devtools_client/condition.js

```diff
@@ -15,17 +16,40 @@ function compile (node) {
     return `(${compile(value)}).length === 0`
   } else if (type === 'isDefined') {
     return `(() => {
-        try {
-          ${value}
-          return true
-        } catch {
-          return false
-        }
-      })()`
+      try {
+        ${value}
+        return true
+      } catch {
+        return false
+      }
+    })()`
   } else if (type === 'instanceof') {
-    return `${compile(value[0])} instanceof ${value[1]}`
+    // TODO: Consider just calling Function.prototype[Symbol.hasInstance] directly
+    return `((klass) => {
+      if (
+        klass &&
+        typeof klass[Symbol.hasInstance] === 'function' &&
+        klass[Symbol.hasInstance] !== Function.prototype[Symbol.hasInstance]
+      ) {
+        throw new Error('Posibility of side effect')
+      } else {
+        return ${compile(value[0])} instanceof klass
+      }
+    })(${value[1]})`
   } else if (type === 'len' || type === 'count') {
-    return `(${compile(value)}).length`
+    return `((val) => {
+      if (
+        typeof val === 'string' ||
+        Array.isArray(val) ||
+        val instanceof Object.getPrototypeOf(Int8Array)
+      ) {
+        return val.length
+      } else if (val instanceof Set || val instanceof Map) {
+        return ${throwOnSideEffect('val', '"size"')}
+      } else {
+        throw new TypeError('Variable does not support len/count')
+      }
+    })(${compile(value)})`
```

Comment:
nice! love how you solved the compile multiple times.

Reply:
Thank you 😊